core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1278,7 +1278,8 @@ class ReplicaManager(val config: KafkaConfig,
           // NotLeaderForPartitionException or ReplicaNotAvailableException.
           // If it is from a follower then send the offset metadata but not the records data as that can be fetched
           // from the remote store.
-          if (remoteLogManager.isDefined && log != null && !log.config.compact) {
+          if (remoteLogManager.isDefined && log != null && !log.config.compact &&
+            log.rlmEnabled && log.config.remoteStorageEnable) {
```

Comment:
It seems these condition checks are duplicate of the Log.remoteLogEnabled method.
Can we call that method instead?